### DISCUSSION TITLE
reveal.js Speaker View for OBS presentations

### DISCUSSION THREAD

**Discussion by UUoocl** [hakimel/reveal.js#3606] [Ideas]
I've written a plug-in that makes it possible to use the reveal.js Speaker View in OBS.  

OBS Studio is a free and open source software for video recording and live streaming. OBS can play reveal.js slides in its embedded browser. One limitation is that the OBS Browser doesn't allow pop up windows. This means the reveal.js Speaker View can't open 😢

The Reveal Slides for OBS plugin modifies the Speaker View so that a pop up window is not needed 🙌. The Speaker View and Slides are added to OBS as separate Browser Sources and WebSocket messages keep them in sync.

[Reveal Slides for OBS](https://github.com/UUoocl/Reveal-Slides-for-OBS) Please try it out.  There is more work to do.  

Any feedback and test results are appreciated.

**Comment by AxelGanter** [comment]
so cool :1st_place_medal:

**Comment by UUoocl** [comment]
A new version of this idea it out!  

The project page is here [Slides Studio](https://uuoocl.github.io/03-tutorials/slidesstudio-obsidian/)

Slides Studio is an HTML page that enables Reveal.js presentations to control Open Broadcast Studio (OBS). 

Create slides in HTML, Markdown or with the reveal.js GUI [Slides.com](http://slides.com/)

Slide Studio innovates on the Reveal.js "[speaker view](https://revealjs.com/speaker-view/)" by adding 3 features.

- Connect to the OBS WebSocket Server to remotely control OBS
- a new "Studio" view to configure OBS settings.
     - Use the Studio view to set the appearance of the slides and camera in OBS
- a new "Teleprompter" view
     - Slide notes are scrolled like a teleprompter
     - with adjustable font size and scroll speed
